File: Capital/下單API_2.1.22/DotNet範例(VS2010)/OrderTesterVC/OverseaFutureProduct.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace OrderTester
{
    public partial class OverseaFutureProduct : Form
    {

        public OverseaFutureProduct()
        {
            InitializeComponent();
        }

        private void OverseaFutureProduct_Load(object sender, EventArgs e)
        {
            //Get Oversea Product
            int m_nCode;

            FOnGetBSTR fOnOverseaFuture = new FOnGetBSTR(OnOverseaFuture);
            m_nCode = Functions.RegisterOnOverseaFuturesCallBack(fOnOverseaFuture);
            m_nCode = Functions.GetOverseaFutures();
        }

        public void OnOverseaFuture(string strMsg)
        {
            DataGridViewRowCollection rows = GridProducts.Rows;

            string[] strRowData = strMsg.Split(';');
            rows.Add(strRowData);
        }
    }
}

File: Capital/下單API_2.1.22/DotNet範例(VS2010)/OrderTesterVB/OverseaFutureProduct.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Public Class OverseaFutureProduct

    Private Sub OverseaFutureProduct_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Dim m_nCode As Integer

        Dim fOnOverseaFuture As New FOnGetBSTR(AddressOf OnOverseaFuture)
        m_nCode = Functions.RegisterOnOverseaFuturesCallBack(fOnOverseaFuture)
        m_nCode = Functions.GetOverseaFutures()

    End Sub

    Public Sub OnOverseaFuture(ByVal strMsg As String)
        Dim rows As DataGridViewRowCollection = GridProducts.Rows

        Dim strRowData As String() = strMsg.Split(";"c)
        rows.Add(strRowData)
    End Sub
End Class